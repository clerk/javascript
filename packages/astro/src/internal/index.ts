/**
 * The following code will be used in order to be injected as script via the astro integration.
 * F.e.
 *
 * injectScript('before-hydration', `...`)
 */

import { createClerkInstance } from './create-clerk-instance';
import { createInjectionScriptRunner } from './create-injection-script-runner';

const runInjectionScript = createInjectionScriptRunner(createClerkInstance);

export { runInjectionScript };

export { generateSafeId } from './utils/generateSafeId';
