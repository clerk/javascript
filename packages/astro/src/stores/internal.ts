import type {
  ActiveSessionResource,
  Clerk,
  ClientResource,
  InitialState,
  OrganizationResource,
  UserResource,
} from '@clerk/types';
import { atom, map } from 'nanostores';

export const $csrState = map<{
  isLoaded: boolean;
  client: ClientResource | undefined | null;
  user: UserResource | undefined | null;
  session: ActiveSessionResource | undefined | null;
  organization: OrganizationResource | undefined | null;
}>({
  isLoaded: false,
  client: null,
  user: null,
  session: null,
  organization: null,
});

export const $initialState = map<InitialState>();

// Use atom instead of `map` to prohibit key changes and allow only replacing the whole object
export const $clerk = atom<Clerk | null>(null);
