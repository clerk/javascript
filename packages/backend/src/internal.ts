export { constants } from './constants';
export { createRedirect } from './createRedirect';
export type { RedirectFun } from './createRedirect';

export type { CreateAuthenticateRequestOptions } from './tokens/factory';
export { createAuthenticateRequest } from './tokens/factory';

export { debugRequestState } from './tokens/request';

export type { AuthenticateRequestOptions, OrganizationSyncOptions } from './tokens/types';

export type { SignedInAuthObjectOptions, SignedInAuthObject, SignedOutAuthObject } from './tokens/authObjects';
export { makeAuthObjectSerializable, signedOutAuthObject, signedInAuthObject } from './tokens/authObjects';

export { AuthStatus } from './tokens/authStatus';
export type { RequestState, SignedInState, SignedOutState } from './tokens/authStatus';

export { decorateObjectWithResources, stripPrivateDataFromObject } from './util/decorateObjectWithResources';

export { createClerkRequest } from './tokens/clerkRequest';
export type { ClerkRequest } from './tokens/clerkRequest';

export { reverificationError, reverificationErrorResponse } from '@clerk/shared/authorization-errors';
