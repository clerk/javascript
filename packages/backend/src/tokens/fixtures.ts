export const mockJwt =
  'eyJhbGciOiJSUzI1NiIsImtpZCI6Imluc18yR0lvUWhiVXB5MGhYN0IyY1ZrdVRNaW5Yb0QiLCJ0eXAiOiJKV1QifQ.eyJhenAiOiJodHRwczovL2FjY291bnRzLmluc3BpcmVkLnB1bWEtNzQubGNsLmRldiIsImV4cCI6MTY2NjY0ODMxMCwiaWF0IjoxNjY2NjQ4MjUwLCJpc3MiOiJodHRwczovL2NsZXJrLmluc3BpcmVkLnB1bWEtNzQubGNsLmRldiIsIm5iZiI6MTY2NjY0ODI0MCwic2lkIjoic2Vzc18yR2JEQjRlbk5kQ2E1dlMxenBDM1h6Zzl0SzkiLCJzdWIiOiJ1c2VyXzJHSXBYT0VwVnlKdzUxcmtabjlLbW5jNlN4ciJ9.n1Usc-DLDftqA0Xb-_2w8IGs4yjCmwc5RngwbSRvwevuZOIuRoeHmE2sgCdEvjfJEa7ewL6EVGVcM557TWPW--g_J1XQPwBy8tXfz7-S73CEuyRFiR97L2AHRdvRtvGtwR-o6l8aHaFxtlmfWbQXfg4kFJz2UGe9afmh3U9-f_4JOZ5fa3mI98UMy1-bo20vjXeWQ9aGrqaxHQxjnzzC-1Kpi5LdPvhQ16H0dPB8MHRTSM5TAuLKTpPV7wqixmbtcc2-0k6b9FKYZNqRVTaIyV-lifZloBvdzlfOF8nW1VVH_fx-iW5Q3hovHFcJIULHEC1kcAYTubbxzpgeVQepGg';

export const mockJwtHeader = {
  alg: 'RS256',
  kid: 'ins_2GIoQhbUpy0hX7B2cVkuTMinXoD',
  typ: 'JWT',
};

export const mockJwtPayload = {
  azp: 'https://accounts.inspired.puma-74.lcl.dev',
  exp: 1666648310,
  iat: 1666648250,
  iss: 'https://clerk.inspired.puma-74.lcl.dev',
  nbf: 1666648240,
  sid: 'sess_2GbDB4enNdCa5vS1zpC3Xzg9tK9',
  sub: 'user_2GIpXOEpVyJw51rkZn9Kmnc6Sxr',
};

export const mockRsaJwkKid = 'ins_2GIoQhbUpy0hX7B2cVkuTMinXoD';

export const mockRsaJwk = {
  use: 'sig',
  kty: 'RSA',
  kid: mockRsaJwkKid,
  alg: 'RS256',
  n: 'u0tNUitBZmcGYMWcqvaRBaJe0XmTQ738RHYoHjhYANyeOkysuu4L_Rqr-fmTXsbebrTp7_OewIqsJXImEWB_WQ3HN9lAkOMCCGDU1udsz_sl1Kwy5JZ7x8Nr4ghXJagQzEF0Ovsj7_TPsBJGkVJ-OiZsTXCe7EAmG5gNGGPBE5Gu14Rwb-eZ5r9RCAaPfhxR1yHYTAvCrku_6i2os7RLpT6UockKtX4QQSH2CMveNwqd6LdwhV8USZrczB2VYkAImngJC745-EWek1sVExYkqheGvC3J8O7D9H4JtaKD2zaq0rJzsIU0zb_wwax5-La-uRuPYvTXlO8B8IK4jjNMCQ',
  e: 'AQAB',
};

export const mockJwks = {
  keys: [mockRsaJwk],
};
