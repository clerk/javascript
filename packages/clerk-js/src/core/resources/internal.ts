export type { default as Clerk } from '../clerk';
export * from './Base';
export * from './UserSettings';
export * from './AuthConfig';
export * from './Client';
export * from './DeletedObject';
export * from './DisplayConfig';
export * from './EmailAddress';
export * from './Environment';
export * from './Error';
export * from './ExternalAccount';
export * from './IdentificationLink';
export * from './Image';
export * from './PhoneNumber';
export * from './Organization';
export * from './OrganizationInvitation';
export * from './OrganizationMembership';
export * from './SamlAccount';
export * from './Session';
export * from './SessionWithActivities';
export * from './SignIn';
export * from './SignUp';
export * from './Token';
export * from './TOTP';
export * from './User';
export * from './Verification';
export * from './Web3Wallet';
