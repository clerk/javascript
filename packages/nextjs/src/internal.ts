/**
 * These need to be explicitly listed. Do not use an * here.
 * If you do, app router will break.
 */
export { MultisessionAppSupport } from './client-boundary/controlComponents';
