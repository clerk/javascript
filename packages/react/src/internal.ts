import type { ErrorThrowerOptions } from '@clerk/shared/error';

import { errorThrower } from './errors/errorThrower';

/**
 * Overrides options of the internal errorThrower (eg setting packageName prefix).
 *
 * @internal
 */
export function setErrorThrowerOptions(options: ErrorThrowerOptions) {
  errorThrower.setMessages(options).setPackageName(options);
}
