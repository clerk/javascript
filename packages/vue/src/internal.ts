export { setErrorThrowerOptions } from './errors/errorThrower';
