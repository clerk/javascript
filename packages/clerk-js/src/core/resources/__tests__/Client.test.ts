import type { ClientJSON, ClientJSONSnapshot } from '@clerk/types';

import { createSession, createSignIn, createSignUp, createUser } from '../../test/fixtures';
import { BaseResource, Client } from '../internal';

describe('Client Singleton', () => {
  it('destroy', async () => {
    const user = createUser({ first_name: 'John', last_name: 'Doe', id: 'user_1' });
    const session = createSession({ id: 'session_1' }, user);
    const clientObjectJSON: ClientJSON = {
      object: 'client',
      id: 'test_id',
      status: 'active',
      last_active_session_id: 'test_session_id',
      sign_in: createSignIn({ id: 'test_sign_in_id' }, user),
      sign_up: createSignUp({ id: 'test_sign_up_id' }), //  This is only for testing purposes, this will never happen
      sessions: [session],
      created_at: jest.now() - 1000,
      updated_at: jest.now(),
    };

    const destroyedSession = createSession(
      {
        id: 'test_session_id',
        abandon_at: jest.now(),
        status: 'ended',
        last_active_token: undefined,
      },
      user,
    );

    const clientObjectDeletedJSON = {
      id: 'test_id_deleted',
      status: 'ended',
      last_active_session_id: null,
      sign_in: null,
      sign_up: null,
      sessions: [destroyedSession],
      created_at: jest.now() - 1000,
      updated_at: jest.now(),
    };

    // @ts-expect-error This is a private method that we are mocking
    BaseResource._fetch = jest.fn().mockReturnValue(
      Promise.resolve({
        client: null,
        response: clientObjectDeletedJSON,
      }),
    );

    const client = Client.getOrCreateInstance().fromJSON(clientObjectJSON);
    expect(client.sessions.length).toBe(1);
    expect(client.createdAt).not.toBeNull();
    expect(client.updatedAt).not.toBeNull();
    expect(client.lastActiveSessionId).not.toBeNull();
    expect(client.signUp.id).toBe('test_sign_up_id');
    expect(client.signIn.id).toBe('test_sign_in_id');

    await client.destroy();

    expect(client.sessions.length).toBe(0);
    expect(client.createdAt).toBeNull();
    expect(client.updatedAt).toBeNull();
    expect(client.lastActiveSessionId).toBeNull();
    expect(client.signUp.id).toBeUndefined();
    expect(client.signIn.id).toBeUndefined();

    // @ts-expect-error This is a private method that we are mocking
    expect(BaseResource._fetch).toHaveBeenCalledWith({
      method: 'DELETE',
      path: `/client`,
    });
  });

  it('has the same initial properties', () => {
    const clientJSON = {
      object: 'client',
      status: null,
      id: 'client_DUMMY_ID',
      sessions: [],
      sign_up: {
        object: 'sign_up',
        id: '',
        status: null,
        required_fields: [],
        optional_fields: [],
        missing_fields: [],
        unverified_fields: [],
        verifications: {
          email_address: {
            object: 'verification',
            id: '',
            status: null,
            strategy: null,
            nonce: null,
            message: null,
            external_verification_redirect_url: null,
            attempts: null,
            expire_at: 1733924546849,
            error: { code: '', message: '', meta: {} },
            verified_at_client: null,
            next_action: '',
            supported_strategies: [],
          },
          phone_number: {
            object: 'verification',
            id: '',
            status: null,
            strategy: null,
            nonce: null,
            message: null,
            external_verification_redirect_url: null,
            attempts: null,
            expire_at: 1733924546849,
            error: { code: '', message: '', meta: {} },
            verified_at_client: null,
            next_action: '',
            supported_strategies: [],
          },
          web3_wallet: {
            object: 'verification',
            id: '',
            status: null,
            strategy: null,
            nonce: null,
            message: null,
            external_verification_redirect_url: null,
            attempts: null,
            expire_at: 1733924546849,
            error: { code: '', message: '', meta: {} },
            verified_at_client: null,
            next_action: '',
            supported_strategies: [],
          },
          external_account: {
            object: 'verification',
            id: '',
            status: null,
            strategy: null,
            nonce: null,
            message: null,
            external_verification_redirect_url: null,
            attempts: null,
            expire_at: 1733924546849,
            error: { code: '', message: '', meta: {} },
            verified_at_client: null,
          },
        },
        username: null,
        first_name: null,
        last_name: null,
        email_address: null,
        phone_number: null,
        has_password: false,
        unsafe_metadata: {},
        created_session_id: null,
        created_user_id: null,
        abandon_at: null,
        web3_wallet: null,
        legal_accepted_at: null,
      },
      sign_in: {
        object: 'sign_in',
        id: '',
        status: null,
        supported_identifiers: [],
        supported_first_factors: [],
        supported_second_factors: null,
        first_factor_verification: {
          object: 'verification',
          id: '',
          status: null,
          strategy: null,
          nonce: null,
          message: null,
          external_verification_redirect_url: null,
          attempts: null,
          expire_at: 1733924546849,
          error: { code: '', message: '', meta: {} },
          verified_at_client: null,
        },
        second_factor_verification: {
          object: 'verification',
          id: '',
          status: null,
          strategy: null,
          nonce: null,
          message: null,
          external_verification_redirect_url: null,
          attempts: null,
          expire_at: 1733924546849,
          error: { code: '', message: '', meta: {} },
          verified_at_client: null,
        },
        identifier: null,
        created_session_id: null,
        user_data: {},
      },
      last_active_session_id: null,
      cookie_expires_at: null,
      created_at: 1733924546843,
      updated_at: 1733924546843,
    } as unknown as ClientJSONSnapshot;

    // @ts-expect-error We cannot mess with the singleton when tests are running in parallel
    const client = new Client(clientJSON);

    expect(client).toMatchSnapshot();
  });

  it('__internal_toSnapshot()', () => {
    const clientJSON = {
      object: 'client',
      status: null,
      id: 'client_DUMMY_ID',
      sessions: [],
      sign_up: {
        object: 'sign_up',
        id: '',
        status: null,
        required_fields: [],
        optional_fields: [],
        missing_fields: [],
        unverified_fields: [],
        verifications: {
          email_address: {
            object: 'verification',
            id: '',
            status: null,
            strategy: null,
            nonce: null,
            message: null,
            external_verification_redirect_url: null,
            attempts: null,
            expire_at: 1733924546849,
            error: { code: '', message: '', meta: {} },
            verified_at_client: null,
            next_action: '',
            supported_strategies: [],
          },
          phone_number: {
            object: 'verification',
            id: '',
            status: null,
            strategy: null,
            nonce: null,
            message: null,
            external_verification_redirect_url: null,
            attempts: null,
            expire_at: 1733924546849,
            error: { code: '', message: '', meta: {} },
            verified_at_client: null,
            next_action: '',
            supported_strategies: [],
          },
          web3_wallet: {
            object: 'verification',
            id: '',
            status: null,
            strategy: null,
            nonce: null,
            message: null,
            external_verification_redirect_url: null,
            attempts: null,
            expire_at: 1733924546849,
            error: { code: '', message: '', meta: {} },
            verified_at_client: null,
            next_action: '',
            supported_strategies: [],
          },
          external_account: {
            object: 'verification',
            id: '',
            status: null,
            strategy: null,
            nonce: null,
            message: null,
            external_verification_redirect_url: null,
            attempts: null,
            expire_at: 1733924546849,
            error: { code: '', message: '', meta: {} },
            verified_at_client: null,
          },
        },
        username: null,
        first_name: null,
        last_name: null,
        email_address: null,
        phone_number: null,
        has_password: false,
        unsafe_metadata: {},
        created_session_id: null,
        created_user_id: null,
        abandon_at: null,
        web3_wallet: null,
        legal_accepted_at: null,
      },
      sign_in: {
        object: 'sign_in',
        id: '',
        status: null,
        supported_identifiers: [],
        supported_first_factors: [],
        supported_second_factors: null,
        first_factor_verification: {
          object: 'verification',
          id: '',
          status: null,
          strategy: null,
          nonce: null,
          message: null,
          external_verification_redirect_url: null,
          attempts: null,
          expire_at: 1733924546849,
          error: { code: '', message: '', meta: {} },
          verified_at_client: null,
        },
        second_factor_verification: {
          object: 'verification',
          id: '',
          status: null,
          strategy: null,
          nonce: null,
          message: null,
          external_verification_redirect_url: null,
          attempts: null,
          expire_at: 1733924546849,
          error: { code: '', message: '', meta: {} },
          verified_at_client: null,
        },
        identifier: null,
        created_session_id: null,
        user_data: {},
      },
      last_active_session_id: null,
      cookie_expires_at: null,
      created_at: 1733924546843,
      updated_at: 1733924546843,
    } as unknown as ClientJSONSnapshot;

    // @ts-expect-error We cannot mess with the singleton when tests are running in parallel
    const client = new Client(clientJSON);

    expect(client.__internal_toSnapshot()).toMatchSnapshot();
  });
});
