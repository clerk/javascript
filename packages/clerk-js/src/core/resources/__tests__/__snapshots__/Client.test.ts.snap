// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Client Singleton has the same initial properties 1`] = `
Client {
  "cookieExpiresAt": null,
  "createdAt": 2024-12-11T13:42:26.843Z,
  "id": "client_DUMMY_ID",
  "lastActiveSessionId": null,
  "pathRoot": "/client",
  "sessions": [],
  "signIn": SignIn {
    "attemptFirstFactor": [Function],
    "attemptSecondFactor": [Function],
    "authenticateWithCoinbaseWallet": [Function],
    "authenticateWithMetamask": [Function],
    "authenticateWithOKXWallet": [Function],
    "authenticateWithPasskey": [Function],
    "authenticateWithRedirect": [Function],
    "authenticateWithWeb3": [Function],
    "create": [Function],
    "createEmailLinkFlow": [Function],
    "createdSessionId": null,
    "firstFactorVerification": Verification {
      "attempts": null,
      "error": {
        "code": "",
        "longMessage": undefined,
        "message": "",
        "meta": {
          "emailAddresses": undefined,
          "identifiers": undefined,
          "paramName": undefined,
          "sessionId": undefined,
          "zxcvbn": undefined,
        },
      },
      "expireAt": 2024-12-11T13:42:26.849Z,
      "externalVerificationRedirectURL": null,
      "message": null,
      "nonce": null,
      "pathRoot": "",
      "status": null,
      "strategy": null,
      "verifiedAtClient": null,
      "verifiedFromTheSameClient": [Function],
    },
    "id": "",
    "identifier": null,
    "pathRoot": "/client/sign_ins",
    "prepareFirstFactor": [Function],
    "prepareSecondFactor": [Function],
    "resetPassword": [Function],
    "secondFactorVerification": Verification {
      "attempts": null,
      "error": {
        "code": "",
        "longMessage": undefined,
        "message": "",
        "meta": {
          "emailAddresses": undefined,
          "identifiers": undefined,
          "paramName": undefined,
          "sessionId": undefined,
          "zxcvbn": undefined,
        },
      },
      "expireAt": 2024-12-11T13:42:26.849Z,
      "externalVerificationRedirectURL": null,
      "message": null,
      "nonce": null,
      "pathRoot": "",
      "status": null,
      "strategy": null,
      "verifiedAtClient": null,
      "verifiedFromTheSameClient": [Function],
    },
    "status": null,
    "supportedFirstFactors": [],
    "supportedIdentifiers": [],
    "supportedSecondFactors": null,
    "userData": UserData {
      "firstName": undefined,
      "hasImage": undefined,
      "imageUrl": undefined,
      "lastName": undefined,
    },
    "validatePassword": [Function],
  },
  "signUp": SignUp {
    "abandonAt": null,
    "attemptEmailAddressVerification": [Function],
    "attemptPhoneNumberVerification": [Function],
    "attemptVerification": [Function],
    "attemptWeb3WalletVerification": [Function],
    "authenticateWithCoinbaseWallet": [Function],
    "authenticateWithMetamask": [Function],
    "authenticateWithOKXWallet": [Function],
    "authenticateWithRedirect": [Function],
    "authenticateWithWeb3": [Function],
    "create": [Function],
    "createEmailLinkFlow": [Function],
    "createdSessionId": null,
    "createdUserId": null,
    "emailAddress": null,
    "firstName": null,
    "hasPassword": false,
    "id": "",
    "lastName": null,
    "legalAcceptedAt": null,
    "missingFields": [],
    "optionalFields": [],
    "pathRoot": "/client/sign_ups",
    "phoneNumber": null,
    "prepareEmailAddressVerification": [Function],
    "preparePhoneNumberVerification": [Function],
    "prepareVerification": [Function],
    "prepareWeb3WalletVerification": [Function],
    "requiredFields": [],
    "status": null,
    "unsafeMetadata": {},
    "unverifiedFields": [],
    "update": [Function],
    "username": null,
    "validatePassword": [Function],
    "verifications": SignUpVerifications {
      "emailAddress": SignUpVerification {
        "attempts": null,
        "error": {
          "code": "",
          "longMessage": undefined,
          "message": "",
          "meta": {
            "emailAddresses": undefined,
            "identifiers": undefined,
            "paramName": undefined,
            "sessionId": undefined,
            "zxcvbn": undefined,
          },
        },
        "expireAt": 2024-12-11T13:42:26.849Z,
        "externalVerificationRedirectURL": null,
        "message": null,
        "nextAction": "",
        "nonce": null,
        "pathRoot": "",
        "status": null,
        "strategy": null,
        "supportedStrategies": [],
        "verifiedAtClient": null,
        "verifiedFromTheSameClient": [Function],
      },
      "externalAccount": Verification {
        "attempts": null,
        "error": {
          "code": "",
          "longMessage": undefined,
          "message": "",
          "meta": {
            "emailAddresses": undefined,
            "identifiers": undefined,
            "paramName": undefined,
            "sessionId": undefined,
            "zxcvbn": undefined,
          },
        },
        "expireAt": 2024-12-11T13:42:26.849Z,
        "externalVerificationRedirectURL": null,
        "message": null,
        "nonce": null,
        "pathRoot": "",
        "status": null,
        "strategy": null,
        "verifiedAtClient": null,
        "verifiedFromTheSameClient": [Function],
      },
      "phoneNumber": SignUpVerification {
        "attempts": null,
        "error": {
          "code": "",
          "longMessage": undefined,
          "message": "",
          "meta": {
            "emailAddresses": undefined,
            "identifiers": undefined,
            "paramName": undefined,
            "sessionId": undefined,
            "zxcvbn": undefined,
          },
        },
        "expireAt": 2024-12-11T13:42:26.849Z,
        "externalVerificationRedirectURL": null,
        "message": null,
        "nextAction": "",
        "nonce": null,
        "pathRoot": "",
        "status": null,
        "strategy": null,
        "supportedStrategies": [],
        "verifiedAtClient": null,
        "verifiedFromTheSameClient": [Function],
      },
      "web3Wallet": SignUpVerification {
        "attempts": null,
        "error": {
          "code": "",
          "longMessage": undefined,
          "message": "",
          "meta": {
            "emailAddresses": undefined,
            "identifiers": undefined,
            "paramName": undefined,
            "sessionId": undefined,
            "zxcvbn": undefined,
          },
        },
        "expireAt": 2024-12-11T13:42:26.849Z,
        "externalVerificationRedirectURL": null,
        "message": null,
        "nextAction": "",
        "nonce": null,
        "pathRoot": "",
        "status": null,
        "strategy": null,
        "supportedStrategies": [],
        "verifiedAtClient": null,
        "verifiedFromTheSameClient": [Function],
      },
    },
    "web3wallet": null,
  },
  "updatedAt": 2024-12-11T13:42:26.843Z,
}
`;

exports[`Client Singleton __internal_toSnapshot() 1`] = `
{
  "cookie_expires_at": null,
  "created_at": 1733924546843,
  "id": "client_DUMMY_ID",
  "last_active_session_id": null,
  "object": "client",
  "sessions": [],
  "sign_in": {
    "created_session_id": null,
    "first_factor_verification": {
      "attempts": null,
      "error": {
        "code": "",
        "long_message": undefined,
        "message": "",
        "meta": {
          "email_addresses": undefined,
          "identifiers": undefined,
          "param_name": undefined,
          "session_id": undefined,
          "zxcvbn": undefined,
        },
      },
      "expire_at": 1733924546849,
      "external_verification_redirect_url": null,
      "id": "",
      "message": null,
      "nonce": null,
      "object": "verification",
      "status": null,
      "strategy": null,
      "verified_at_client": null,
    },
    "id": "",
    "identifier": null,
    "object": "sign_in",
    "second_factor_verification": {
      "attempts": null,
      "error": {
        "code": "",
        "long_message": undefined,
        "message": "",
        "meta": {
          "email_addresses": undefined,
          "identifiers": undefined,
          "param_name": undefined,
          "session_id": undefined,
          "zxcvbn": undefined,
        },
      },
      "expire_at": 1733924546849,
      "external_verification_redirect_url": null,
      "id": "",
      "message": null,
      "nonce": null,
      "object": "verification",
      "status": null,
      "strategy": null,
      "verified_at_client": null,
    },
    "status": null,
    "supported_first_factors": [],
    "supported_identifiers": [],
    "supported_second_factors": null,
    "user_data": {
      "first_name": undefined,
      "has_image": null,
      "image_url": null,
      "last_name": undefined,
    },
  },
  "sign_up": {
    "abandon_at": null,
    "created_session_id": null,
    "created_user_id": null,
    "email_address": null,
    "external_account": undefined,
    "external_account_strategy": undefined,
    "first_name": null,
    "has_password": false,
    "id": "",
    "last_name": null,
    "legal_accepted_at": null,
    "missing_fields": [],
    "object": "sign_up",
    "optional_fields": [],
    "phone_number": null,
    "required_fields": [],
    "status": null,
    "unsafe_metadata": {},
    "unverified_fields": [],
    "username": null,
    "verifications": {
      "email_address": {
        "attempts": null,
        "error": {
          "code": "",
          "long_message": undefined,
          "message": "",
          "meta": {
            "email_addresses": undefined,
            "identifiers": undefined,
            "param_name": undefined,
            "session_id": undefined,
            "zxcvbn": undefined,
          },
        },
        "expire_at": 1733924546849,
        "external_verification_redirect_url": null,
        "id": "",
        "message": null,
        "next_action": "",
        "nonce": null,
        "object": "verification",
        "status": null,
        "strategy": null,
        "supported_strategies": [],
        "verified_at_client": null,
      },
      "external_account": {
        "attempts": null,
        "error": {
          "code": "",
          "long_message": undefined,
          "message": "",
          "meta": {
            "email_addresses": undefined,
            "identifiers": undefined,
            "param_name": undefined,
            "session_id": undefined,
            "zxcvbn": undefined,
          },
        },
        "expire_at": 1733924546849,
        "external_verification_redirect_url": null,
        "id": "",
        "message": null,
        "nonce": null,
        "object": "verification",
        "status": null,
        "strategy": null,
        "verified_at_client": null,
      },
      "phone_number": {
        "attempts": null,
        "error": {
          "code": "",
          "long_message": undefined,
          "message": "",
          "meta": {
            "email_addresses": undefined,
            "identifiers": undefined,
            "param_name": undefined,
            "session_id": undefined,
            "zxcvbn": undefined,
          },
        },
        "expire_at": 1733924546849,
        "external_verification_redirect_url": null,
        "id": "",
        "message": null,
        "next_action": "",
        "nonce": null,
        "object": "verification",
        "status": null,
        "strategy": null,
        "supported_strategies": [],
        "verified_at_client": null,
      },
      "web3_wallet": {
        "attempts": null,
        "error": {
          "code": "",
          "long_message": undefined,
          "message": "",
          "meta": {
            "email_addresses": undefined,
            "identifiers": undefined,
            "param_name": undefined,
            "session_id": undefined,
            "zxcvbn": undefined,
          },
        },
        "expire_at": 1733924546849,
        "external_verification_redirect_url": null,
        "id": "",
        "message": null,
        "next_action": "",
        "nonce": null,
        "object": "verification",
        "status": null,
        "strategy": null,
        "supported_strategies": [],
        "verified_at_client": null,
      },
    },
    "web3_wallet": null,
  },
  "status": null,
  "updated_at": 1733924546843,
}
`;
